### ISSUE TITLE
Task is not being removed from tasklist after checked

### ISSUE THREAD

**Issue by dtracydepp** [nss-day-cohort-46/How-to-Ask-for-Help#116]
**_Ticket MUST be completed before getting help from an instructor_**
#### Have you searched the other issue tickets? If not, do that first. 
- [x ] Yes
- If No, stop and search.

####  REQUIRED Add Tags to Issue Ticket and Assign to Project Board
 

- [ ] Added to `c46 Help-Thread` project board
- [ ] Added the `Blocker` label if this is stoping you from moving forward

#### REQUIRED Describe your issue:
>> When I check the checkbox on the task, it is changing from false to true in the database but the task is not being removed from the DOM or the database

#### REQUIRED Copy/paste any error messages that appear:

N/A
#### REQUIRED What have you googled? **_(You must have at least 3 links related to your issue)_**
- [ ] 1. https://www.freecodecamp.org/news/react-change-inline-css-conditionally-based-on-component-state/
- [ ] 2. https://reactjs.org/docs/forms.html
- [ ] 3. https://medium.com/@tariqul.islam.rony/multiple-checkbox-handling-by-react-js-84b1d49a46c6

#### REQUIRED: What have you tried? Tell us everything you have tried. 
>> added .filter, added new state components, changed the .map

#### REQUIRED: Copy/paste link TO THE FILE on your branch that you are having issues with

**Your link:https://github.com/nss-day-cohort-46/reactive-nutshell-shield-of-swiftness-1/blob/87ac639cf071d364cddaf88d1a723028801ed790/src/components/tasks/TaskList.js

https://github.com/nss-day-cohort-46/reactive-nutshell-shield-of-swiftness-1/blob/894ac9e33423b7b31b4a77e371f4f126a4c0ef93/src/components/tasks/Task.js
**

#### Code Snippet

```
export const TaskList = () => {
    const history = useHistory()
    // This state changes when `getTasks()` is invoked below
    const { tasks, getTasks} = useContext(TaskContext)
    const [filteredTasks, setFilteredTasks] = useState([])
    
     //useEffect - reach out to the world for something - API call for the tasks; wil only run one time at intial render because array is empty
    useEffect(() => {
      getTasks()
      }, [])

   
    useEffect(() => {
      getTasks()
      .then(()=>{
        const subset = filteredTasks.filter(task=> task.isComplete === false)
        setFilteredTasks(subset)
      })

    }, [])


    return (
      <div className="tasks">
          <h2>Tasks</h2>
          <button onClick={() => {history.push("/tasks/create")}}>
            Add New Task
          </button>
        {/* {console.log("TaskList: Render", tasks)} */}
        {
            // using .map method to iterate the array of tasks and generate HTML for each one by invoking the "Task" comp
          tasks.map(taskObj=> {
            //   key and task become properties on the object passed in as in argument
            return <Task key={taskObj.id} task={taskObj} />
          })
        }
      </div>
    )
  } 
```

**Comment by jakefroeb** [comment]
so this problem is occuring because your useEffect that filters your tasks by whether they are complete or not only runs once after render. in your second useEffect on line 22 try something like 
```
useEffect(()=>{
        const subset = filteredTasks.filter(task=> task.isComplete === false)
        setFilteredTasks(subset)
},[tasks])
```
this will will change your filtered tasks every time your tasks variable is changed, and because when you complete a task it runs getTasks() at the end it should trigger this useEffect which will update your filteredTasks and cause a rerender with the new tasks with the newly completed task and the new filteredTasks which will omit the newly completed task.

**Comment by jisie** [comment]
@dtracydepp Are you still having this issue?

**Comment by dtracydepp** [comment]
useEffect(() => {
    const subset = tasks.filter(task => task.isComplete === false)
    setFilteredTasks(subset)

  }, [tasks])

I was calling getTasks twice after initial render and 
I should have been filtering through tasks instead of filteredTasks and I also didn't have a .then(getTasks) in my provider to addTasks
